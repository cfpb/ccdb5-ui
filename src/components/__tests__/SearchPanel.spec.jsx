import ReduxSearchPanel, { SearchPanel } from '../Search/SearchPanel'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore( {
    aggs: {
      lastIndexed: new Date( '2016-02-01T05:00:00.000Z' )
    },
    query: {
      date_received_max: '2020-04-05T12:00:00.000Z',
      date_received_min: '2011-12-31T12:00:00.000Z',
      searchField: 'all',
      searchText: 'something searching',
    }
  })

  return renderer.create(
    <Provider store={ store } >
      <IntlProvider locale="en">
        <ReduxSearchPanel />
      </IntlProvider>
    </Provider>
  )
}

describe('component:SearchPanel', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
