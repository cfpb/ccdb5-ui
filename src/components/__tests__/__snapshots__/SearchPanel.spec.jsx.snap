// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:SearchPanel renders without crashing 1`] = `
<div
  className="search-panel"
>
  <h2>
    Search complaint data 
    <span
      className="date-subscript"
    >
      (last updated: 
      <span>
        2/1/2016
      </span>
      )
    </span>
  </h2>
  <div>
    <div
      className="search-bar"
      role="search"
    >
      <form
        action=""
        onSubmit={[Function]}
      >
        <h3
          className="h4"
        >
          Search within
        </h3>
        <div
          className="layout-row"
        >
          <div
            className="cf-select flex-fixed"
          >
            <select
              aria-label="Choose which field will be searched"
              id="searchField"
              onChange={[Function]}
              value="all"
            >
              <optgroup
                label="Search Within"
              >
                <option
                  value="all"
                >
                  All data
                </option>
                <option
                  value="company"
                >
                  Company name
                </option>
                <option
                  value="complaint_what_happened"
                >
                  Narratives
                </option>
              </optgroup>
            </select>
          </div>
          <div
            className="flex-all typeahead-portal"
          >
            <section
              className="typeahead "
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="m-btn-inside-input input-contains-label"
              >
                <div
                  className="input-contains-label_before input-contains-label_before__search"
                >
                  <svg
                    className="cf-icon-svg"
                    viewBox="0 0 880.6 1200"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M860.1 879.2L645.7 664.8c90.8-136.8 76-323-44.6-443.5-137.5-137.5-360.5-137.5-498 0s-137.5 360.5 0 498c118.5 118.4 303.9 137 443.5 44.6L761 978.3c27.3 27.3 71.7 27.3 99 0s27.4-71.8.1-99.1zm-508-116.9C191.4 762 61.3 631.5 61.6 470.8c.3-160.7 130.8-290.8 291.5-290.5s290.8 130.8 290.5 291.5c-.2 118.2-71.9 224.6-181.5 269.1-34.9 14.2-72.3 21.5-110 21.4z"
                    />
                  </svg>
                </div>
                <label
                  className="u-visually-hidden"
                  htmlFor="searchText"
                >
                  Enter the term you want to search for
                </label>
                <input
                  autoComplete="off"
                  className="a-text-input"
                  id="searchText"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Enter your search term(s)"
                  type="text"
                  value="something searching"
                />
                <button
                  className="a-btn a-btn__link"
                  onClick={[Function]}
                >
                  <svg
                    className="cf-icon-svg"
                    viewBox="0 0 718.9 1200"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
                    />
                  </svg>
                  Clear
                </button>
              </div>
            </section>
          </div>
          <button
            className="a-btn flex-fixed"
            type="submit"
          >
            Search
          </button>
          <a
            className="u-visually-hidden"
            href="#search-summary"
          >
            Skip to Results
          </a>
          <div
            className="advanced-container flex-fixed"
          >
            <button
              className="a-btn a-btn__link"
              onClick={[Function]}
            >
              Show\x20
              advanced search tips
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
